### ISSUE TITLE
Words ending in -er sometimes return an incorrect split

### ISSUE THREAD

**Issue by ibanner56** [nlp-compromise/nlp-syllables#8]
ex:
"sooner" => ["sooner"]
"later" => ["later"]

"hunter" => ["hun", "ter"]
"donner" => ["don", "ner"]

**Comment by spencermountain** [comment]
ooh! good find. thanks Isaac, will check it out.

**Comment by yclicc** [comment]
There seems to be a more general issue with "er" or "ne":
"energy"=>["ener","gy"]
"onerous"=>["one","rous"]
"oneiric"=>["onei","ric"]

but not
"oneself"=>["one","self"]
"one-shot"=>["one-","shot"]

**Comment by spencermountain** [comment]
hey, i took a look at this this week, but didn't see an easy fix. I'm curious, what are you two using this for? I'm pretty beat-up with [comprimise@11](https://twitter.com/nlp_compromise/status/878287627266342912)  right now and may not have the time for this, PRs welcomed though.

**Comment by yclicc** [comment]
I'm trying to use it as a pre-processing step before the automatic pointing of Psalm (or other) texts ([see here](https://en.wikipedia.org/wiki/Anglican_chant)) via a recurrent neural network (I actually ported your code to Python which I will probably publish soon with your permission). It seems like a dead end though as the syllabification the algorithm produces doesn't match [my test set](http://spellerweb.net/cmindex/Chant/Psalms.html) in any case. Currently researching more reinforcement learning style techniques (as a correct pointing will always satisfy certain conditions).

I'd love to submit a pr if I came across one but as noted [here](http://www.syllablecount.com/syllable/rules/) the English language has a lot of edge cases.

**Comment by spencermountain** [comment]
oh, very cool @yclicc!
yeah, i'd be happy to help, and feel-free to release anything, in any form.

To be honest, the issue I ran into with this wasn't the # of edge-cases, (which is resolvable with some work) but the actual splitting of the letters stopped making sense. 

`bourgeois` -> `bour` `geo` `is`  and stuff.

like, the best way to model this is to have pronunciation, then represent it as an array of phenomes, or whatever. There are a few [metaphone](https://en.wikipedia.org/wiki/Metaphone) or soundex libraries, that return actual pronounciation, which is more useful to have when it comes to syllables, really.

**Comment by spencermountain** [comment]
i think what i'm saying is that if I were to do this again, i'd just implement metaphone3, then write some syllable-rules with that output. that's a more clever way to do this

**Comment by ibanner56** [comment]
I was using this to try and detect haikus in a discord chat, but I actually switched over to just doing a lookup against the Moby Project's hyphenated language data:

http://icon.shef.ac.uk/Moby/

I'm still falling back on another syllable calculator package if the word isn't in the database, but that one is also filled with bugs related to edge cases in the language...

**Comment by spencermountain** [comment]
oh hey, that's cool.
hey lets build this, you two.
to be perfectly honest, i'm not sure why compromise is a dependency at all. `.split(' ')` would do just fine, unless we need sentence-splitting too?

maybe this library should just be string-in, array-out, then people can do whatever they want with it..

yeah, learning rules from that moby file is already making my heart race, i'm not sure i can not-do it now..

**Comment by ibanner56** [comment]
Yeah, I wasn't sure what compromise brought to the table here, but I haven't used compromise in the past.

I've already got a version of the database parsed out into json (words mapped to syllable arrays), which is probably more helpful than the original hyphenated data.

**Comment by ibanner56** [comment]
http://www.isaacbanner.com/realtalk/syllables.json

(large file - you're probably better downloading it first, rather than opening it in a browser)

**Comment by spencermountain** [comment]
hey @ibanner56 wanna diff it with the current results, to see the % mistakes it gets right now?

i guess if we had a list of all the mistakes, we could sort the errors by-syllable, to see what's patterns are the worst? like, i bet it's "er" and "ne" ?
¯\_(ツ)_/¯

**Comment by ibanner56** [comment]
While this also brought to focus a few problems in the parsed Moby data, here's a diff of all words where Moby and your output disagree on syllable count (you split some syllables differently, so there are a lot more results if you look for an absolute match over a syllable count match):

http://isaacbanner.com/realtalk/syllable_diff.json

(again, large file)

**Comment by spencermountain** [comment]
hey, been busy, but haven't forgotten about this.
